src/app/dynamic/enterprise/quotas/component.ts

```diff
@@ -100,11 +103,41 @@ export class QuotasComponent implements OnInit {
     this.dataSource.filter = query;
   }
 
-  async add(): Promise<void> {
-    await lastValueFrom(
+  add(): void {
+    this._openDialog(null);
+  }
+
+  edit(quota: QuotaDetails): void {
+    this._openDialog(quota);
+  }
+
+  async delete({name}: QuotaDetails): Promise<void> {
+    const isConfirmed = await lastValueFrom(
+      this._matDialog
+        .open<ConfirmationDialogComponent, ConfirmationDialogConfig, boolean>(ConfirmationDialogComponent, {
+          data: {
+            title: 'Delete Quota',
+            message: `Delete quota for ${name}?`,
+            confirmLabel: 'Delete',
+          },
+        })
+        .afterClosed()
+    );
+
+    if (!isConfirmed) return;
+
+    await lastValueFrom(this._quotaService.deleteQuota(name));
+
+    this._notificationService.success(`Deleting the ${name} datacenter`);
```

Comment:
Showing **undefined** and we are not deleting **datacenter**

https://user-images.githubusercontent.com/17727069/179679572-eb458f43-e7f6-4912-8c8a-9204fff1646c.mp4